bazel/BUILD.bazel

```diff
@@ -79,6 +79,17 @@ ic_version_or_git_sha(
     visibility = ["//visibility:public"],
 )
 
+genrule(
+    name = "commit_timestamp_txt",
+    outs = ["commit_timestamp.txt"],
+    cmd = "git show -s --format=%ct > $@",
+    tags = [
+        "local",
+        "no-cache",
+    ],
+    visibility = ["//visibility:public"],
+)
```

Comment:
The commit date isn't really the build time (for instance you can, today, build an image from a commit from a year ago). I'm missing some context as to how this is used exactly, but is this required on every build, or only on release builds?

Would it make sense, instead of reading the local repo and introducing some (minor) impurity, to pass the build time as a flag to the build when it is necessary? This will avoid rebuilds and ensure the build time is injected only when actually necessary

Reply:
Right. Similar to how we handle [embedding the IC_VERSION git commit](https://sourcegraph.com/search?q=repo:%5Egithub%5C.com/dfinity/ic%24+ic_version_rc_only&patternType=keyword&sm=0).

Reply:
Wasn't aware of that, but yes, exactly like that! Thanks for the link

Reply:
I introduced a new `release_build` flag. This allows us to implement the non-release default behavior in the target specicially for the use-case instead of defining it the common bazel script. For example, in this case, we can calculate the commit time in release builds but use a fake commit time far in the future in non-release builds. Other targets may require other default values so I believe this approach gives us more flexibility and requires less propagation of values through multiple layers. Please let me know what you think.

> The commit date isn't really the build time

I renamed build-time to commit-time for clarity.